back/routes/adventure.js

```diff
@@ -1,70 +1,115 @@
-require("dotenv/config");
-const {Parse} = require('./../parse');
+require('dotenv/config');
+const { Parse } = require('./../parse');
 var express = require('express');
 var router = express.Router();
 var axios = require('axios');
 
 const api = process.env.NODE_ENV_GOOGLE_MAPS_API_KEY;
 
-// ----- Get restaurants ------
-router.post('/restaurants', async(req, res) => {
-  const query = "restaurant";
-  const location = `${req.body.lat},${req.body.lng}`;
-  const radius = 1000;
-  const url = encodeURI(`https://maps.googleapis.com/maps/api/place/textsearch/json?query=${query}&location=${location}&radius=${radius}&key=${process.env.NODE_ENV_GOOGLE_MAPS_API_KEY}`)
+// ----- Get Google API restaurants -----
+async function getGoogleRestaurants(radius, lat, lng) {
+  const query = 'restaurant';
+  const location = `${lat},${lng}`;
+  const url = encodeURI(
+    `https://maps.googleapis.com/maps/api/place/textsearch/json?query=${query}&location=${location}&radius=${radius}&key=${process.env.NODE_ENV_GOOGLE_MAPS_API_KEY}`,
+  );
   var config = {
     method: 'get',
     url: url,
-    headers: { }
+    headers: {},
   };
-  axios(config)
-  .then(function (response) {
-    res.status(200).send(response.data.results);
-  })
-})
+  const results = axios(config).then(function (response) {
+    return response.data.results;
+  });
+  return results;
+}
 
 // Find an user's preference
-async function preferenceQuery(username){
+async function preferenceQuery(username) {
   const query = new Parse.Query('UserPreference');
-  query.equalTo("username", username);
+  query.equalTo('username', username);
   const currentPreference = await query.first();
-  return currentPreference.toJSON()
+  return currentPreference.toJSON();
 }
 
+// Find mean for reviews of restaurant on a feedback
+async function getMean(experienceId, feedbackId) {
+  const values = await new Parse.Query('Review')
+    .equalTo('experienceId', experienceId)
+    .equalTo('feedbackId', feedbackId)
+    .select('score')
```

Comment:
I wonder if you can call `select` twice so we don't need to read the JSON field below? Either way, thanks for using the `select` syntax!

Reply:
Thank you for letting me know! Really appreciate it